src/cmap/connection.ts

```diff
@@ -468,6 +489,8 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
     this.removeAllListeners(Connection.PINNED);
     this.removeAllListeners(Connection.UNPINNED);
 
+    this[kMessageStream].destroy();
+
     options = Object.assign({ force: false }, options);
     if (this[kStream] == null || this.destroyed) {
       this.destroyed = true;
```

Comment:
optional - let's clean up a bit!  `this[kStream]` is non-nullable ( see line 194 ).

We can remove the `null` check here (I'm surprised Typescript didn't complain..?), and then line 473 is unnecessary.